programs/psa/psa_hash.c

```diff
@@ -0,0 +1,163 @@
+        psa_hash_abort(&cloned_psa_hash_operation);
+        return EXIT_FAILURE;
+    }
+
+    /* Compute hash using multi-part operation */
+
+    status = psa_hash_setup(&psa_hash_operation, HASH_ALG);
+    if (status != PSA_SUCCESS) {
+        mbedtls_printf("psa_hash_setup failed\n");
+        psa_hash_abort(&psa_hash_operation);
+        psa_hash_abort(&cloned_psa_hash_operation);
+        return EXIT_FAILURE;
+    }
+
+    status = psa_hash_update(&psa_hash_operation, buf, sizeof(buf));
+    if (status != PSA_SUCCESS) {
+        mbedtls_printf("psa_hash_update failed\n");
+        psa_hash_abort(&psa_hash_operation);
+        psa_hash_abort(&cloned_psa_hash_operation);
+        return EXIT_FAILURE;
+    }
+
+    status = psa_hash_clone(&psa_hash_operation, &cloned_psa_hash_operation);
+    if (status != PSA_SUCCESS) {
+        mbedtls_printf("PSA hash clone failed");
+        psa_hash_abort(&psa_hash_operation);
+        psa_hash_abort(&cloned_psa_hash_operation);
+        return EXIT_FAILURE;
+    }
+
+    status = psa_hash_finish(&psa_hash_operation, hash, sizeof(hash), &hash_size);
+    if (status != PSA_SUCCESS) {
+        mbedtls_printf("psa_hash_finish failed\n");
+        psa_hash_abort(&psa_hash_operation);
+        psa_hash_abort(&cloned_psa_hash_operation);
+        return EXIT_FAILURE;
+    }
+
+    status =
+        psa_hash_verify(&cloned_psa_hash_operation, mbedtls_test_sha256_hash,
+                        mbedtls_test_sha256_hash_len);
+    if (status != PSA_SUCCESS) {
+        mbedtls_printf("psa_hash_verify failed\n");
+        psa_hash_abort(&psa_hash_operation);
+        psa_hash_abort(&cloned_psa_hash_operation);
+        return EXIT_FAILURE;
+    } else {
+        mbedtls_printf("Multi-part hash operation successful!\n");
+    }
+
+    /* Compute hash using one-shot function call */
+    memset(hash, 0, sizeof(hash));
+    hash_size = 0;
+
+    status = psa_hash_compute(HASH_ALG,
+                              buf, sizeof(buf),
+                              hash, sizeof(hash),
+                              &hash_size);
+    if (status != PSA_SUCCESS) {
+        mbedtls_printf("psa_hash_compute failed\n");
+        psa_hash_abort(&psa_hash_operation);
+        psa_hash_abort(&cloned_psa_hash_operation);
+        return EXIT_FAILURE;
+    }
+
+    for (size_t j = 0; j < mbedtls_test_sha256_hash_len; j++) {
+        if (hash[j] != mbedtls_test_sha256_hash[j]) {
+            mbedtls_printf("One-shot hash operation failed!\n\n");
+            psa_hash_abort(&psa_hash_operation);
+            psa_hash_abort(&cloned_psa_hash_operation);
+            return EXIT_FAILURE;
+        }
+    }
+
+    mbedtls_printf("One-shot hash operation successful!\n\n");
+
+    mbedtls_printf("The SHA-256( '%s' ) is:\n", buf);
+
+    for (size_t j = 0; j < mbedtls_test_sha256_hash_len; j++) {
+        if (j % 8 == 0) {
+            mbedtls_printf("\n    ");
+        }
+        mbedtls_printf("%02x ", hash[j]);
+    }
```

Comment:
I think we should just dump the hex without spaces and line breaks. That's how online calculators typically do it.